### ISSUE TITLE
git+ssh dependencies are being pruned / removed incorrectly.

### ISSUE THREAD

**Issue by kevingilbert100** [npm/npm#19596]
#### I'm opening this issue because:

  - [ ] npm is crashing.
  - [x] npm is producing an incorrect install.
  - [ ] npm is doing something I don't understand.
  - [ ] Other (_see below for feature requests_):

#### What's going wrong?

In NPM 5.6.0 save a git+ssh repo to package.json dependencies. Do a npm install... git+ssh dependencies will be there and be listed in package.json, do a npm prune, the packages will be deleted from the node_modules directory but still exist inside the package.json file.

```
kgilbert:osio-slave kgilbert$ npm i

> dtrace-provider@0.8.6 install /Users/kgilbert/Projects/overspeed/osio-slave/node_modules/dtrace-provider
> node-gyp rebuild || node suppress-error.js

  ACTION binding_gyp_ndtp_target_build_ndtp .
  TOUCH Release/obj.target/ndtp.stamp
added 17 packages in 13.781s
kgilbert:osio-slave kgilbert$ cd node_modules/osio-config/
kgilbert:osio-config kgilbert$ cd ../../
kgilbert:osio-slave kgilbert$ npm prune
removed 17 packages in 8.823s
kgilbert:osio-slave kgilbert$ cd node_modules/osio-config
-bash: cd: node_modules/osio-config: No such file or directory
kgilbert:osio-slave kgilbert$ cat package.json
...
  "dependencies": {
    "osio-config": "git+ssh://git@github.com/OverSpeedIO/osio-config.git",
    "osio-log": "git+ssh://git@github.com/OverSpeedIO/osio-log.git",
    ....
  },
...
kgilbert:osio-slave kgilbert$
```

Please note the ```...``` inside the package.json file does not exist in the real package.json file and this is just to remove sensitive information in this post.

**Comment by Romasato** [comment]
Yeah, seeing the same: all my git+ssh packages in `package.json` get removed after `npm install <somePackage>` and this issue already has multiple tickets created. Not sure if development team is aware of it at all?

**Comment by brentonhouse** [comment]
This might be the same or related issue:  #19527

**Comment by G-Rath** [comment]
Ya, I think this is another duplication of my issue: https://github.com/npm/npm/issues/19394

Sadly the development team doesn't seem to have actioned it yet :(

**Comment by yedlosh** [comment]
And also this one https://github.com/npm/npm/issues/17379

**Comment by jsynowiec** [comment]
Duplicate of #19356 (I think it's the oldest/first one).

**Comment by Rauttis** [comment]
Ran into the same issue, git+ssh dependencies configured under `dependencies` in `package.json` are being pruned.

Weirdly enough this issue is happening in a build that has been using npm 5.4.2 for a while now without any issues. Does npm hit a server while doing npm prune?

**Comment by rmja** [comment]
There is also #16839 which has evolved into the same issue.
